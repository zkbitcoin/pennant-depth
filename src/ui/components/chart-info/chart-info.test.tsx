import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { ChartInfo } from "./chart-info";

test("ChartInfo renders correctly", () => {
    // bounds must be a tuple of exactly 2 Dates
    const bounds: [Date, Date] = [new Date(2021, 2, 1), new Date(2021, 2, 2)];

    render(<ChartInfo bounds={bounds} />);

    // Check that the elements exist (React 19-safe async)
    expect(screen.getByText(/01 Mar 2021/)).toBeInTheDocument();
    expect(screen.getByText(/02 Mar 2021/)).toBeInTheDocument();
    expect(screen.getByText(/to/)).toBeInTheDocument();

    // Check wrapper div class using data-testid
    const wrapper = screen.getByTestId("chart-info-wrapper");
    expect(wrapper).toHaveClass("chart-info-wrapper");

    // Snapshot test
    expect(wrapper).toMatchSnapshot();
});
