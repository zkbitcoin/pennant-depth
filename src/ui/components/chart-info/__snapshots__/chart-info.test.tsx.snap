// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartInfo renders correctly 1`] = `
<div
  class="chart-info-wrapper"
  data-testid="chart-info-wrapper"
>
  <div>
    <span>
      00:00 01 Mar 2021
    </span>
    <span
      class="text-muted"
    >
       to 
    </span>
    <span>
      00:00 02 Mar 2021
    </span>
  </div>
</div>
`;
